code/ui/manager/src/components/sidebar/Tree.tsx

```diff
@@ -132,47 +177,67 @@ interface NodeProps {
   docsMode: boolean;
   isOrphan: boolean;
   isDisplayed: boolean;
+  color: string | undefined;
   isSelected: boolean;
   isFullyExpanded?: boolean;
   isExpanded: boolean;
   setExpanded: (action: ExpandAction) => void;
   setFullyExpanded?: () => void;
   onSelectStoryId: (itemId: string) => void;
+  status: State['status'][keyof State['status']];
+  api: API;
 }
 
 const Node = React.memo<NodeProps>(function Node({
   item,
+  status,
   refId,
   docsMode,
   isOrphan,
   isDisplayed,
   isSelected,
   isFullyExpanded,
+  color,
   setFullyExpanded,
   isExpanded,
   setExpanded,
   onSelectStoryId,
+  api,
 }) {
-  const api = useStorybookApi();
-  if (!isDisplayed) return null;
+  if (!isDisplayed) {
+    return null;
+  }
 
   const id = createId(item.id, refId);
   if (item.type === 'story' || item.type === 'docs') {
     const LeafNode = item.type === 'docs' ? DocumentNode : StoryNode;
+
+    const statusIcon = Object.values(status || {}).reduce<
+      typeof status[keyof typeof status]['status']
+    >((acc, s) => {
+      if (order.indexOf(s.status) > order.indexOf(acc)) {
+        return s.status;
+      }
+      return acc;
+    }, 'unknown');
+    const [icon, iconColor, textColor] = mapping[statusIcon];
+
     return (
-      <LeafNodeStyleWrapper>
+      <LeafNodeStyleWrapper
+        data-selected={isSelected}
+        data-ref-id={refId}
+        data-item-id={item.id}
+        data-parent-id={item.parent}
+        data-nodetype={item.type === 'docs' ? 'document' : 'story'}
+        data-highlightable={isDisplayed}
+        className="sidebar-item"
```

Comment:
Is this a breaking change? I assume you do this so the attributes wrap the icon/tooltip. You could alternatively just put the icon/tooltip inside the `DocumentNode`/`StoryNode`?

Reply:
I didn't consider it a breaking change.. I had to make it because the leaf-node is an anchor tag with a href.
I can't put another button or anchor inside, that's invalid.
So I moved these attributes and background styling up to the wrapper, made that `display:flex` and added the icon (it's clickable!) besides the link.